scenarios/kubernetes_e2e.py

```diff
@@ -308,76 +394,7 @@ def main(args):
     if args.stage_suffix:
         runner_args.append('--stage-suffix=%s' % args.stage_suffix)
 
-    cluster = args.cluster or 'e2e-gce-%s-%s' % (
-        os.environ['NODE_NAME'], os.getenv('EXECUTOR_NUMBER', 0))
-
-    if args.kubeadm:
-        # Not from Jenkins
-        cluster = args.cluster or 'e2e-kubeadm-%s' % os.getenv('BUILD_NUMBER', 0)
-
-        # This job only runs against the kubernetes repo, and bootstrap.py leaves the
-        # current working directory at the repository root. Grab the SCM_REVISION so we
-        # can use the .debs built during the bazel-build job that should have already
-        # succeeded.
-        status = re.search(
-            r'STABLE_BUILD_SCM_REVISION ([^\n]+)',
-            check_output('hack/print-workspace-status.sh')
-        )
-        if not status:
-            raise ValueError('STABLE_BUILD_SCM_REVISION not found')
-
-        opt = '--deployment kubernetes-anywhere' \
-            ' --kubernetes-anywhere-path /workspace/kubernetes-anywhere' \
-            ' --kubernetes-anywhere-phase2-provider kubeadm' \
-            ' --kubernetes-anywhere-cluster %s' \
-            ' --kubernetes-anywhere-kubeadm-version' \
-            ' gs://kubernetes-release-dev/bazel/%s/build/debs/' % (cluster, status.group(1))
-            # The gs:// path given here should match jobs/ci-kubernetes-bazel-build.sh
-        mode.add_environment('E2E_OPT=%s' % opt)
-
-    mode.add_environment(
-      # Boilerplate envs
-      # Skip gcloud update checking
-      'CLOUDSDK_COMPONENT_MANAGER_DISABLE_UPDATE_CHECK=true',
-      # Use default component update behavior
-      'CLOUDSDK_EXPERIMENTAL_FAST_COMPONENT_UPDATE=false',
-      # E2E
-      'E2E_UP=%s' % args.up,
-      'E2E_TEST=%s' % args.test,
-      'E2E_DOWN=%s' % args.down,
-      'E2E_NAME=%s' % cluster,
-      # AWS
-      'KUBE_AWS_INSTANCE_PREFIX=%s' % cluster,
-      # GCE
-      'INSTANCE_PREFIX=%s' % cluster,
-      'KUBE_GCE_NETWORK=%s' % cluster,
-      'KUBE_GCE_INSTANCE_PREFIX=%s' % cluster,
-      # GKE
-      'CLUSTER_NAME=%s' % cluster,
-      'KUBE_GKE_NETWORK=%s' % cluster,
-    )
-
-    # env blacklist.
-    # TODO(krzyzacy) change this to a whitelist
-    docker_env_ignore = [
-      'GOOGLE_APPLICATION_CREDENTIALS',
-      'GOPATH',
-      'GOROOT',
-      'HOME',
-      'PATH',
-      'PWD',
-      'WORKSPACE'
-    ]
-
-    # TODO(fejta): delete this
-    mode.add_environment(*(
-        '%s=%s' % (k, v) for (k, v) in os.environ.items()
-        if k not in docker_env_ignore))
-
-    # Overwrite JOB_NAME for soak-*-test jobs
-    if args.soak_test and os.environ.get('JOB_NAME'):
-        mode.add_environment('JOB_NAME=%s' % os.environ.get('JOB_NAME').replace('-test', '-deploy'))
-
+    update_environment(mode, args)
```

Comment:
This change feels too invasive. It makes this code feel more like a complicated program rather than a trivial script.

I realize the docker subclassing adds a lot of complexity however we intend to eliminate all of it in favor of the non-docker version in the near future.